### PR TITLE
feat: dynamic sidebar content

### PR DESCRIPTION
**Pull request by misohu**
Related to this [jira task](https://warthogs.atlassian.net/browse/KF-567?atlOrigin=eyJpIjoiN2M3MjQwZmY2NzEwNGQwMTkzYThkZTNhMzU0YmFlNmUiLCJwIjoiaiJ9) and this [feature](https://github.com/canonical/kubeflow-dashboard-operator/issues/8)

In `kubeflow-dashboard-operator` there is a `sidebar` relation. In order to add element to the sidebar proper relation needs to be established. After the relation is set, charms fills the config for given app-name.

Sidebar element is removed after the relation is broken.

### CODE REVIEW

File: charms/tensorboards-web-app/src/charm.py [hunk 1]
```diff
@@ -130,6 +139,31 @@ def _configure_mesh(self, interfaces):
...
+    def _on_sidebar_relation_joined(self, event: RelationJoinedEvent):
+        if not self.unit.is_leader():
+            return
+        event.relation.data[self.app].update(
+            {
+                "config": json.dumps(
+                    [
+                        {
+                            "app": self.app.name,
+                            "position": 4,
```
Comment (DnPlas) [inline charms/tensorboards-web-app/src/charm.py:151]: I wonder what happens with the position if this is the only existing element in the sidebar. Do you know?

**Review by DnPlas — COMMENTED** [review]
Just a question, other than that LGTM.

File: charms/tensorboards-web-app/src/charm.py (hunk 1 above)
Comment (misohu) [reply charms/tensorboards-web-app/src/charm.py:151]: nothing it will be first and the order of items ion the array will be preserved

**Comment by DnPlas** [comment]
There is a current effort for designing and implementing the dynamic sidebar feature, which will introduce some changes that will likely be incompatible with this PR. 
Closing this PR because of that, but feel free to re-open if needed. Thanks @misohu !

**Pull request closed without merging**